src/sentry/integrations/slack/integration.py

```diff
@@ -44,22 +45,35 @@
     "text": "The Slack integration adds a new Alert Rule action to all projects. To enable automatic notifications sent to Slack you must create a rule using the slack workspace action in your project settings.",
 }
 
+reauthentication_alert = {
+    "alertText": _("Slack must be re-authorized to avoid a disruption of Slack notifications"),
+}
+
 metadata = IntegrationMetadata(
     description=_(DESCRIPTION.strip()),
     features=FEATURES,
     author="The Sentry Team",
     noun=_("Workspace"),
     issue_url="https://github.com/getsentry/sentry/issues/new?title=Slack%20Integration:%20&labels=Component%3A%20Integrations",
     source_url="https://github.com/getsentry/sentry/tree/master/src/sentry/integrations/slack",
-    aspects={"alerts": [setup_alert]},
+    aspects={"alerts": [setup_alert], "reauthentication_alert": reauthentication_alert},
 )
 
 
+class SlackIntegration(IntegrationInstallation):
+    def get_config_data(self):
+        metadata = self.model.metadata
+        # classic bots had a user_access_token in the metadata
+        default_installation = "classic_bot" if "user_access_token" in metadata else "workspace_app"
+        return {"installationType": metadata.get("installation_type", default_installation)}
```

Comment:
How do the `born_as_bot` and `migrated_to_bot` types get created?

Reply:
@markstory Good question. When you do a fresh install of a bot app, we set `installation_type` as `born_as_bot`. That's in the code already. In the migration flow that we are going to build out, we will set `installation_type` as `migrated_to_bot`.